cmd/sriovdp/manager.go

```diff
@@ -208,28 +210,57 @@ func (rm *resourceManager) discoverHostDevices() error {
 			}
 			glog.Infof("discoverDevices(): device found: %-12s\t%-12s\t%-20s\t%-40s", device.Address, device.Class.ID, vendorName, productName)
 
-			// exclude device in-use in host
-			if isDefaultRoute, _ := hasDefaultRoute(device.Address); !isDefaultRoute {
+			// Network devices
+			if devClass == netClass {
+				// exclude device in-use in host
+				if isDefaultRoute, _ := hasDefaultRoute(device.Address); !isDefaultRoute {
 
-				aPF := utils.IsSriovPF(device.Address)
-				aVF := utils.IsSriovVF(device.Address)
+					aPF := utils.IsSriovPF(device.Address)
+					aVF := utils.IsSriovVF(device.Address)
 
-				if aPF || !aVF {
-					// add to linkWatchList
-					rm.addToLinkWatchList(device.Address)
-				}
+					if aPF || !aVF {
+						// add to linkWatchList
+						rm.addToLinkWatchList(device.Address)
+					}
+
+					if aPF && utils.SriovConfigured(device.Address) {
+						// do not add this device in net device list
+						continue
+					}
+
+					if newDevice, err := resources.NewPciNetDevice(device, rm.rFactory); err == nil {
+						rm.netDeviceList = append(rm.netDeviceList, newDevice)
+					} else {
+						glog.Errorf("discoverDevices() error adding new device: %q", err)
+					}
 
-				if aPF && utils.SriovConfigured(device.Address) {
-					// do not add this device in net device list
-					continue
 				}
+			}
+
+			// FEC processing accelerators
+			if devClass == fecClass {
+				//Check if device is a physical function (PF)
+				aPF := utils.IsSriovPF(device.Address)
 
-				if newDevice, err := resources.NewPciNetDevice(device, rm.rFactory); err == nil {
-					rm.netDeviceList = append(rm.netDeviceList, newDevice)
+				//If dev ID = 0b30 it means it is a RSU device and not FEC
+				if device.Product.ID != "0b30" {
```

Comment:
use a well named constant instead of hard-coding a hex in the middle of the code

BTW is this code manufacturer -i.e. Intel- specific?

Reply:
According to [here](https://www.intel.com/content/www/us/en/programmable/documentation/xgz1560360700260.html) it's the PCI device ID for the N3000 FPGA device.